packages/kbn-config-schema/src/types/type.ts

```diff
@@ -83,6 +93,10 @@ export abstract class Type<V> {
     this.internalSchema = schema;
   }
 
+  public extendsDeep(newOptions: ExtendsDeepOptions): Type<V> {
+    return this;
+  }
```

Comment:
>NIT: Type is abstract, so I think we may be able to have Type#extendsDeep be abstract too?

I only have ~7 types extend this function while the remaining (~12) re-use this default implementation. Do you think it would be better to define `extendsDeep` explicitly in each?

Reply:
Yeah, no, having the default noop implementation on the base type seems better then, you're right.

Reply:
Not a problem!